packages/marker/dev/index.tsx

```diff
@@ -0,0 +1,32 @@
+import { Component, createMemo, createSignal } from "solid-js";
+import { render } from "solid-js/web";
+import "uno.css";
+import { createMarker } from "../src";
+
+const LOREM_IPSUM = [
+  "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse ligula tortor, mollis congue augue ac, sagittis sodales odio. Duis vulputate feugiat metus. Curabitur in nisl ac felis vestibulum facilisis vitae eu lectus. In justo metus, viverra non leo quis, laoreet finibus lacus. Phasellus consectetur arcu orci, non varius ligula ornare ac. Nulla facilisi. Maecenas dignissim nulla et sodales venenatis. Nunc purus nunc, consequat vitae convallis sit amet, bibendum quis augue. Pellentesque purus turpis, aliquet eget libero ut, imperdiet fermentum lacus. In sit amet finibus nunc, faucibus tincidunt eros. Praesent pretium lobortis turpis, nec rhoncus dui interdum aliquet. Nulla fermentum tellus nec dui fermentum rhoncus. Ut cursus purus ligula, sed tincidunt diam vulputate sit amet. Sed in eleifend eros.",
+  "Duis nunc massa, mollis ac suscipit in, feugiat sed risus. Morbi imperdiet dapibus felis in congue. Fusce commodo, leo et tristique laoreet, ligula sapien tempus arcu, eget molestie lorem nisl at nulla. Proin dictum bibendum est, ut faucibus neque hendrerit sit amet. Nullam gravida elit sit amet placerat tristique. Morbi ipsum elit, suscipit ac rhoncus id, ornare in turpis. Vivamus convallis placerat est, sit amet dignissim risus rhoncus at. Aenean nec erat et lectus lobortis commodo quis ut eros. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer aliquet tempor mauris, eget commodo enim scelerisque eget. Quisque venenatis porta euismod. Vestibulum at consectetur nibh. Mauris eleifend, dui at consequat fermentum, lorem dolor gravida nibh, vel porta tellus ligula ac risus. Donec convallis, nibh a ultricies imperdiet, velit velit aliquam velit, tempus aliquet lectus purus ac risus.",
+] as const;
+
+const App: Component = () => {
+  const [search, setSearch] = createSignal("");
+
+  const highlight = createMarker(text => <mark>{text()}</mark>);
+  const regex = createMemo(() => new RegExp(search(), "gi"));
```

Comment:
How does this handle inputs that result in illegal regex, like "+" or "]"?

Since the example is also a form of documentation, we should at least add a comment and/or an error boundary.

Reply:
could you explain a bit more? I would love to improve this example (in the our site too), but I don't exactly know how.
Should I just wrap it in a `try catch`? I wanted to search for some quality examples online, but every article is just showing that you can use `new Regex` to create a dynamic regex, not how it's supposed to be done

Reply:
alright, I've added `makeSearchRegex` to improve that part a little bit

Reply:
Good idea.